dbms/src/Interpreters/Join.h

```diff
@@ -192,6 +204,28 @@ class Join
     };
 
 
+    struct ProbeProcessInfo
+    {
+        Block block;
+        UInt64 max_block_size;
+        size_t start_row;
+        size_t end_row;
+        bool all_rows_joined_finish;
+        bool block_full;
+
+        ProbeProcessInfo(Block block_, UInt64 max_block_size_, size_t start_row_, size_t end_row_, bool all_rows_joined_finish_, bool block_full_)
+            : block(block_)
+            , max_block_size(max_block_size_)
+            , start_row(start_row_)
+            , end_row(end_row_)
+            , all_rows_joined_finish(all_rows_joined_finish_)
+            , block_full(block_full_)
+        {}
+    };
+
+    using ProbeProcessInfoPtr = std::shared_ptr<ProbeProcessInfo>;
```

Comment:
Can we use `unique_ptr` and pass `probe_process_info` by reference `ProbeProcessInfo &`?